catalog/spatial/registry/registry-federation-admin-service-impl/src/test/java/org/codice/ddf/registry/federationadmin/service/impl/FederationAdminServiceImplTest.java

```diff
@@ -13,6 +13,1722 @@
+        JAXBElement<RegistryPackageType> jaxbRegistryPackage = RIM_FACTORY.createRegistryPackage(
+                getTestRegistryPackage());
+        when(catalogFramework.query(any(QueryRequest.class))).thenReturn(response);
+        when(parser.unmarshal(any(ParserConfigurator.class),
+                eq(JAXBElement.class),
+                any(InputStream.class))).thenReturn(jaxbRegistryPackage);
+        RegistryPackageType registryPackage = fasi.getRegistryObjectByMetacardId(metacardId, null);
+        assertThat(registryPackage, notNullValue());
+        verify(fasi).getRegistryObjectByMetacardId(metacardId, null);
+    }
+
+    @Test(expected = FederationAdminException.class)
+    public void getRegistryObjectByMetacardIdEmptyString() throws FederationAdminException {
+        assertThat(fasi.getRegistryObjectByMetacardId(""), is(nullValue()));
+    }
+
+    @Test
+    public void getRegistryObjectByMetacardIdThatDoesNotExist() throws Exception {
+        QueryRequest request = getTestQueryRequest();
+        QueryResponse response = getPopulatedTestQueryResponse(request);
+        Metacard metacard = getTestMetacard();
+        when(security.getSystemSubject()).thenReturn(subject);
+        when(catalogFramework.query(any(QueryRequest.class))).thenReturn(response);
+        when(registryTransformer.transform(any(InputStream.class))).thenReturn(metacard);
+        assertThat(fasi.getRegistryObjectByMetacardId("Not a metacard"), is(nullValue()));
+    }
+
+    @Test(expected = FederationAdminException.class)
+    public void getRegistryObjectByMetacardIdDuplicateMetacards() throws Exception {
+        String metacardId = "metacardId";
+        Metacard metacard = getTestMetacard();
+        metacard.setAttribute(new AttributeImpl(Metacard.ID, metacardId));
+        metacard.setAttribute(new AttributeImpl(Metacard.METADATA, "xmlString"));
+        QueryRequest request = getTestQueryRequest();
+        QueryResponse response = getPopulatedTestQueryResponse(request, metacard, metacard);
+        when(security.getSystemSubject()).thenReturn(subject);
+        when(catalogFramework.query(any(QueryRequest.class))).thenReturn(response);
+        fasi.getRegistryObjectByMetacardId(metacardId);
+        verify(fasi).getRegistryObjectByMetacardId(metacardId);
+        verify(catalogFramework).query(any(QueryRequest.class));
+        verify(parser, never()).unmarshal(any(ParserConfigurator.class),
+                eq(JAXBElement.class),
+                any(InputStream.class));
+    }
+
+    @Test
+    public void testRefreshRegistrySubscriptionsNullBundleContext() throws Exception {
+        fasi.refreshRegistrySubscriptions();
+    }
+
+    @Test
+    public void testcreateRemoteEntries() throws Exception {
+        Metacard firstMetacard = getTestMetacard();
+        Metacard secondMetacard = getTestMetacard();
+        Metacard thirdMetacard = getTestMetacard();
+        String registryId = "registryId";
```

Comment:
To @vinamartin's point seems like this should just be a class constant and then you will need to declare it throughout various test methods.